src/ucs/datastruct/lru.h

```diff
@@ -0,0 +1,50 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2001-2023. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef __LRU_H__
+#define __LRU_H__
+
+#include <ucs/type/status.h>
+#include <stddef.h>
+
+typedef struct ucs_lru *ucs_lru_h;
+
+/**
+ * @brief Init a new LRU cache instance.
+ *
+ * @param [in] capacity  Max number of elements to store.
+ *
+ * @return Pointer to the allocated cache. In case of error, NULL returned.
+ */
+ucs_lru_h ucs_lru_init(size_t capacity);

```

Comment:
IMO mixing status & ptr in the same return value is bad.
Better return a status (which perhaps might be "error" in some other additional cases in the future) and the output ptr separately.